source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -102,6 +128,21 @@ void QuicFilterManagerConnectionImpl::adjustBytesToSend(int64_t delta) {
   write_buffer_watermark_simulation_.checkLowWatermark(bytes_to_send_);
 }
 
+void QuicFilterManagerConnectionImpl::maybeHandleDelayedClose() {
```

Comment:
I think this could use better naming.  By name I assumed this would be handling the close(... delayClose) call not applying the delay close policy after a write.
The comments in the .h file are great but if we can make the name more clear that'd be even better.